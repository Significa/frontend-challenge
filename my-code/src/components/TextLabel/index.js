import sc from 'styled-components'
import { theme } from 'styled-tools'

import { text4 } from '../../styles/text'

const TextLabel = sc.div`
  display: block;
  padding: 4px 6px;
  height: 28px;
  box-sizing: border-box;
  border-radius: 4px;
  ${text4}
  font-weight: 500;
  line-height: 20px;
  background-color: ${theme('greyLight')};
  color: ${theme('greyDark')};
  margin-left: 8px;
`

export default TextLabel
