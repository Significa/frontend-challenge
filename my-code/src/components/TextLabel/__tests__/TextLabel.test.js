import React from 'react'
import { shallow } from 'enzyme'

import TextLabel from '../'

describe('TextLabel', () => {
  it('renders correctly', () => {
    const tree = shallow(<TextLabel>label text</TextLabel>)
    expect(tree).toMatchSnapshot()
  })
})
