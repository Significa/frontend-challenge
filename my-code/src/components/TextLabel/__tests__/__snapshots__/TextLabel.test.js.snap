// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLabel renders correctly 1`] = `
.c0 {
  display: block;
  padding: 4px 6px;
  height: 28px;
  box-sizing: border-box;
  border-radius: 4px;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 500;
  line-height: 20px;
  background-color: theme.greyLight;
  color: theme.greyDark;
  margin-left: 8px;
}

<div
  className="c0"
>
  label text
</div>
`;
